internal/services/databricks/databricks_virtual_network_peering_resource_test.go

```diff
@@ -0,0 +1,266 @@
+
+func TestAccDatabricksVirtualNetworkPeering_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_databricks_virtual_network_peering", "test")
+	r := DatabricksVirtualNetworkPeeringResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccDatabricksVirtualNetworkPeering_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_databricks_virtual_network_peering", "test")
+	r := DatabricksVirtualNetworkPeeringResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.RequiresImportErrorStep(r.requiresImport),
+	})
+}
+
+func TestAccDatabricksVirtualNetworkPeering_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_databricks_virtual_network_peering", "test")
+	r := DatabricksVirtualNetworkPeeringResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.update(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccDatabricksVirtualNetworkPeering_complete(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_databricks_virtual_network_peering", "test")
+	r := DatabricksVirtualNetworkPeeringResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccDatabricksVirtualNetworkPeering_completeUpdate(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_databricks_virtual_network_peering", "test")
+	r := DatabricksVirtualNetworkPeeringResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.completeUpdate(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func (DatabricksVirtualNetworkPeeringResource) Exists(ctx context.Context, clients *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := vnetpeering.ParseVirtualNetworkPeeringID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := clients.DataBricks.VnetPeeringClient.Get(ctx, *id)
+	if err != nil {
+		return nil, fmt.Errorf("making Read request on Databricks %s: %+v", *id, err)
```

Comment:
since the Exists function is used for both checking it exists and that it doesn't, we'll want to handle the 404 here when it doesn't, as that's raised as an error (since we're not expecting a 404 by default when calling the `Get` function):

```suggestion
		if response.WasNotFound(resp.HttpResponse) {
			return pointer.To(false), nil
		}
		return nil, fmt.Errorf("making Read request on Databricks %s: %+v", *id, err)
```

Reply:
Fixed.